app.js

```diff
@@ -111,6 +113,18 @@
             method: 'POST',
             json: cred
         }, function (err, res) {
+            if(err) {
+
+                if(err.code == 'ENOTFOUND') {
```

Comment:
Please add a block for ETIMEDOUT as well. The else block will suffice for all other error codes. These 3 seem to be relevant for a client connection.